crates/codegen/parser/runtime/src/templates/mod.rs.jinja2

```diff
@@ -5,7 +5,7 @@ pub mod cst;
 pub mod cursor;
 pub mod kinds;
 pub mod language;
-pub(crate) mod lexer;
+pub mod lexer;
```

Comment:
We need to expose `ScannedToken` as it's a part of the public API now; we should either hide the lexer-related API and tests as unit tests (rather than integration ones) or just expose this module anyway

Reply:
> we should either hide the lexer-related API

Is that the part of #640 about hiding the `scan()` API? I suggest doing so, so that we don't have to expose `ScannedToken` or this machinery at all. It is a rather internal solution that will change (frequently) over time, and I don't think there is any (reported) user benefit for it right now. I think we will need to think of how we can expose this long term once we have embedded language support.

Reply:
> tests as unit tests (rather than integration ones)

I think we have very few lexer-related tests. We can either move them to snapshot tests (testing via a calling parser), or if not possible for some corner case, move it to unit tests like you suggested.

Reply:
Separated in https://github.com/NomicFoundation/slang/pull/728; I'll revert the changes unrelated changes here. The lexer test has to be a unit test, since we directly want to test the immediate behaviour of `next_token`